Switch test results from Trac to Github links

youssefsoli opened the pull request:
#### 16168e138f9bd3ffb0ca218a92366de8447b0a77
<pre>
Switch test results from Trac to Github links
<a href="https://bugs.webkit.org/show_bug.cgi?id=242316">https://bugs.webkit.org/show_bug.cgi?id=242316</a>

Reviewed by Jonathan Bedard.

* LayoutTests/fast/harness/results.html:
* ManualTests/resources/test-results-page.html:

Canonical link: <a href="https://commits.webkit.org/252138@main">https://commits.webkit.org/252138@main</a>
</pre>

youssefsoli commented on LayoutTests/fast/harness/results.html:
```diff
@@ -802,12 +802,9 @@
     {
         let basePath;
-        if (this.shouldUseTracLinks()) {
-            let revision = this.testResults.revision;
```
Not sure if revisions are still attached to test results. I wasn't able to find instances of this so I assume this is a deprecated attribute.

aproskuryakov replied:
Looks like the intention was to link to the exact revision of the test file, as opposed to the latest; but this appears to have broken a long time ago - perhaps with buildbot 2 transition.

youssefsoli replied:
Do we have a translation service like `https://commits.webkit.org/` but for file/tree views in GitHub? I think there would be a decent amount of places where this could be used across the codebase.

JonWBedard replied:
We don't but it wouldn't be particularly difficult to add.

youssefsoli commented on LayoutTests/fast/harness/results.html:
```diff
@@ -1105,8 +1102,8 @@
-            let hasTrac = src.indexOf('trac.webkit.org') != -1;
-            let prefix = hasTrac ? 'trac.webkit.org/.../' : '';
+            let hasGithub = src.indexOf('github.com/WebKit/WebKit/') != -1;
+            let prefix = hasGithub ? 'github.com/WebKit/WebKit/.../' : '';
```
This is just for the iframe label, so the link doesn't need to be functional.

youssefsoli committed:
Switch test results from Trac to Github links
https://bugs.webkit.org/show_bug.cgi?id=242316

Reviewed by Jonathan Bedard.

* LayoutTests/fast/harness/results.html:
* ManualTests/resources/test-results-page.html:

Canonical link: https://commits.webkit.org/252138@main
